### PR TITLE
Painting the fence task

### PR DESCRIPTION
**Pull request by olexandr-maltsev-work**

### CODE REVIEW

File: src/main/java/com/softserve/algorithms/Algorithms.java [hunk 1]
```diff
@@ -54,4 +55,28 @@ else if (i < m)
...
     }
+
+    /**
+     *
+     * @param n number of posts
+     * @param k number of colors
+     * @return count of ways to color n posts using k colors
+     */
+    public static BigDecimal countWaysPaintingTheFence(int n, int k) {
+        if (n < 1 || k < 1) {
```
Comment (KatyaKaratnyk) [inline src/main/java/com/softserve/algorithms/Algorithms.java]: please, give meaningful names for variables

**Review by KatyaKaratnyk — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/algorithms/Algorithms.java (hunk 1 above)
Comment (olexandr-maltsev-work) [reply src/main/java/com/softserve/algorithms/Algorithms.java]: Done

**Review by KatyaKaratnyk — APPROVED** [review]

**Review by KatyaKaratnyk — APPROVED** [review]

**Review by KatyaKaratnyk — APPROVED** [review]

**Review by KatyaKaratnyk — APPROVED** [review]

**Review by KatyaKaratnyk — APPROVED** [review]

File: src/main/java/com/softserve/algorithms/Algorithms.java [hunk 2]
```diff
@@ -54,4 +55,28 @@ else if (i < m)
...
         }
     }
+
+    /**
+     *
+     * @param posts - number of posts
+     * @param colors - number of colors
+     * @return count of ways to color n posts using k colors
+     */
+    public static BigDecimal countWaysPaintingTheFence(int posts, int colors) {
```
Comment (igorflyunt) [inline src/main/java/com/softserve/algorithms/Algorithms.java]: https://github.com/Java364/AlgorithmsCLI/blob/c352bc875dc33eb9d3bb84f393e62627a3502e0b/src/main/java/com/softserve/algorithms/Algorithms.java#L63-L65
Add `throws` clause to your method and add `@throws` tag to method's javadoc.

**Review by igorflyunt — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/algorithms/Algorithms.java (hunk 2 above)
Comment (olexandr-maltsev-work) [reply src/main/java/com/softserve/algorithms/Algorithms.java]: Added `throws` and changed javadoc

**Review by igorflyunt — APPROVED** [review]

**Review by igorflyunt — APPROVED** [review]

**Pull request merged**